cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/LdapResourceCRLFetcher.java

```diff
@@ -0,0 +1,223 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.adaptors.x509.authentication.handler.support;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jasig.cas.util.CompressionUtils;
+import org.ldaptive.Connection;
+import org.ldaptive.ConnectionConfig;
+import org.ldaptive.ConnectionFactory;
+import org.ldaptive.DefaultConnectionFactory;
+import org.ldaptive.LdapAttribute;
+import org.ldaptive.LdapEntry;
+import org.ldaptive.LdapException;
+import org.ldaptive.Response;
+import org.ldaptive.ResultCode;
+import org.ldaptive.SearchOperation;
+import org.ldaptive.SearchRequest;
+import org.ldaptive.SearchResult;
+import org.ldaptive.pool.AbstractConnectionPool;
+import org.ldaptive.pool.ConnectionPool;
+import org.ldaptive.pool.PooledConnectionFactory;
+import org.ldaptive.provider.Provider;
+import org.springframework.core.io.ByteArrayResource;
+
+import javax.naming.AuthenticationException;
+import javax.naming.OperationNotSupportedException;
+import javax.validation.constraints.NotNull;
+import java.security.cert.CertificateException;
+import java.security.cert.X509CRL;
+
+/**
+ * Fetches a CRL from an LDAP instance.
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public class LdapResourceCRLFetcher extends ResourceCRLFetcher {
+
+    /** Attribute name in LDAP to indicate the cert revocation list. **/
+    public static final String DEFAULT_CERTIFICATE_REVOCATION_LIST_ATTRIBUTE = "certificateRevocationList;binary";
+
+    /** The Certificate revocation list attribute name.*/
+    protected final String certificateRevocationListAttributeName;
+
+    /** The connection configuration to ldap.*/
+    protected final ConnectionConfig connectionConfig;
+
+    /** Search request to find the attribute in the ldap tree.*/
+    protected final SearchRequest searchRequest;
+
+    /** The ldap provider, defaults to {@link DefaultConnectionFactory#getProvider()}. */
+    protected Provider provider = DefaultConnectionFactory.getDefaultProvider();
+
+    /** The connection factory to prep for connections. **/
+    protected ConnectionFactory connectionFactory;
+
+    private AbstractConnectionPool connectionPool;
+
+    /**
+     * Instantiates a new Ldap resource cRL fetcher.
+     * initializes the default certificate revocation attribute
+     * to be {@link #DEFAULT_CERTIFICATE_REVOCATION_LIST_ATTRIBUTE}.
+     * @param searchRequest the search request
+     * @param connectionConfig the connection config
+     */
+    public LdapResourceCRLFetcher(@NotNull final SearchRequest searchRequest,
+                                  @NotNull final ConnectionConfig connectionConfig) {
```

Comment:
I think I ran into a few problems with that approach. A factory requires a connection-config to spin up and that config must have an ldap url before it can be used. Since I dont know the ldap url beforehand, I opted to simply pass the config around. The config needs to be prepped dynamically so it can be set on the factory.